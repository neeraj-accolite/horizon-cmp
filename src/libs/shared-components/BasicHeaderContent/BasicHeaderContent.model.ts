export interface BasicHeaderContentProps {
  titleText: string;
  descriptionText: string;
  titleClassName?: string;
  descriptionClassName?: string;
}
