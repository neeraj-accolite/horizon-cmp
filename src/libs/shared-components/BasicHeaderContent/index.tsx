export { default as TitleDescription } from './BasicHeaderContent';

export type { BasicHeaderContentProps } from './BasicHeaderContent.model';
