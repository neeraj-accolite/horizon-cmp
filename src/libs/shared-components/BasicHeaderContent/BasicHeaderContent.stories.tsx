import { Meta, StoryObj } from '@storybook/react';
import BasicHeaderContent from './BasicHeaderContent';
export default {
  title: 'Components/BasicHeaderContent',
  component: BasicHeaderContent,
  parameters: { layout: 'centered' },
} as Meta<typeof BasicHeaderContent>;

type Story = StoryObj<typeof BasicHeaderContent>;

export const Default: Story = {
  args: {
    titleText: 'Default Title',
    descriptionText: 'This is the default description.',
  },
};

export const WithCustomStyles: Story = {
  args: {
    titleText: 'Styled Title',
    descriptionText: 'This description uses custom styles.',
    titleClassName: 'text-blue-500 text-lg',
    descriptionClassName: 'text-gray-700 text-sm',
  },
};

export const LongText: Story = {
  args: {
    titleText: 'Long Title Example',
    descriptionText:
      'This is an example of a long description. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin id bibendum metus, non commodo enim.',
  },
};

export const ShortText: Story = {
  args: {
    titleText: 'Short Title',
    descriptionText: 'Short description.',
  },
};
