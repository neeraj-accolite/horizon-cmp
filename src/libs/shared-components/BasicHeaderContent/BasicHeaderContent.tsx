import React from 'react';
import { BasicHeaderContentProps } from './BasicHeaderContent.model';

const TitleDescription: React.FC<BasicHeaderContentProps> = ({
  titleText,
  descriptionText,
  titleClassName = '',
  descriptionClassName = '',
}) => {
  return (
    <div className={`flex flex-col gap-[2.125rem] sm:gap-10`}>
      <h2 className={`${titleClassName}`}>{titleText}</h2>
      <p className={`${descriptionClassName}`}>{descriptionText}</p>
    </div>
  );
};

export default TitleDescription;
