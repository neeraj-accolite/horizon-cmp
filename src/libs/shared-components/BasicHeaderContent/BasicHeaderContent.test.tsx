import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import BasicHeaderContent from './BasicHeaderContent';

test('renders title and description correctly', () => {
  const { getByText } = render(
    <BasicHeaderContent
      titleText="Test Title"
      descriptionText="Test Description"
    />,
  );

  const titleElement = getByText('Test Title');
  const descriptionElement = getByText('Test Description');

  expect(titleElement).toBeInTheDocument();
  expect(descriptionElement).toBeInTheDocument();
});

test('applies custom class names to title and description', () => {
  const { getByText } = render(
    <BasicHeaderContent
      titleText="Custom Title"
      descriptionText="Custom Description"
      titleClassName="custom-title-class"
      descriptionClassName="custom-description-class"
    />,
  );

  const titleElement = getByText('Custom Title');
  const descriptionElement = getByText('Custom Description');

  expect(titleElement).toHaveClass('custom-title-class');
  expect(descriptionElement).toHaveClass('custom-description-class');
});

test('applies default gap styles and responsive adjustments', () => {
  const { container } = render(
    <BasicHeaderContent
      titleText="Test Title"
      descriptionText="Test Description"
    />,
  );

  const containerElement = container.firstChild;

  expect(containerElement).toHaveClass(
    'flex',
    'flex-col',
    'gap-[2.125rem]',
    'sm:gap-10',
  );
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <BasicHeaderContent
      titleText="Snapshot Title"
      descriptionText="Snapshot Description"
    />,
  );

  expect(asFragment()).toMatchSnapshot();
});
