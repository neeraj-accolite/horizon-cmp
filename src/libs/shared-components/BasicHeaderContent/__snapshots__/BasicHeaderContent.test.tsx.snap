// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-[2.125rem] sm:gap-10"
  >
    <h2
      class=""
    >
      Snapshot Title
    </h2>
    <p
      class=""
    >
      Snapshot Description
    </p>
  </div>
</DocumentFragment>
`;
